BuildaGitServer/Base/Authentication.swift

```diff
@@ -40,9 +40,17 @@ extension ProjectAuthenticator: KeychainStringSerializable {
         
         let comps = value.componentsSeparatedByString(":")
         guard comps.count >= 4 else { throw Error.withInfo("Corrupted keychain string") }
-        guard let service = GitService(rawValue: comps[0]) else {
-            throw Error.withInfo("Unsupported service: \(comps[0])")
+
+        var service: GitService
+        switch comps[0] {
+        case GitService.GitHub.hostname():
+            service = GitService.GitHub
+        case GitService.BitBucket.hostname():
+            service = GitService.BitBucket
+        default:
+            service = GitService.EnterpriseGitHub(host: comps[0])
```

Comment:
Hmm this will be tricky when someone tries to add a service not on any supported server. Could you instead try to synchronously ping the server and detect whether it really is a GitHub Enterprise server? We'll need this anyway to support more self-hosted servers like BitBucket Server - we'll need to ping the parsed host and detect which type it is. So using the `default` branch of the switch is here might cause issues with error reporting (we'll let users add even completely unsupported servers, which I'd like to avoid).